apps/store/extensions/app/social-reviews/modules/reviews-utils.js

```diff
@@ -43,37 +46,71 @@ var ReviewUtils = {};
         var usernameOnReview;
+        var myIndex;
         for (var index = 0; index < reviews.length; index++) {
             review = reviews[index];
             usernameOnReview = review.actor.id;
             review.actor.id = cleanUsername(review.actor.id);
+            review.isMyComment = (usernameOnReview === formatUsername(user));
             //Only populate review details if there is a logged in
             //user
-            if (user) {
-                review.iLike = socialSvc.isUserliked(formattedUsername, review.object.id, 1);;
-                review.iDislike = socialSvc.isUserliked(formattedUsername, review.object.id, 0);;
-                review.isMyComment = (usernameOnReview === formatUsername(user)) ? true : false;
+            if (user && review.isMyComment) {
+                myIndex = index;
             }
         }
-        return reviews;
+        if (myIndex !== undefined) {
+            reviews.splice(myIndex, 1);
+        }
+        return {'allReviews': reviews, 'myReview': myReview};
     };
     ReviewUtils.listReviews = function(target, user, userPagination) {
         var socialSvc = getSocialSvc();
         var paging = buildPagination(userPagination || {});
         var obj = JSON.parse(String(socialSvc.getSocialObjectJson(target, paging.sortBy, paging.offset, paging.limit)));
-        return processUserReviews(socialSvc, user, obj.attachments || []);
+        return processUserReviews(socialSvc, user, obj.attachments || [], target);
     };
-    ReviewUtils.createUserReview = function(review) {
+    ReviewUtils.updateUserReview = function (review) {
         var socialSvc = getSocialSvc();
         var reviewJSON = JSON.stringify(review);
-        var id = socialSvc.publish(reviewJSON);
+        var updatedReview = JSON.parse(String(socialSvc.update(reviewJSON)));
+        updatedReview = isEmpty(updatedReview) ? null : updatedReview;
+        if (updatedReview) {
+            updatedReview.actor.id = cleanUsername(updatedReview.actor.id);
+        }
+        return updatedReview;
+    };
+    ReviewUtils.createUserReview = function (review) {
+        var socialSvc = getSocialSvc();
+        var reviewJSON = JSON.stringify(review);
+        var username = review.actor.id;
+        var target = review.target.id;
+        var reviewed = socialSvc.isReviewed(target, username);
         var result = {};
+        var id = -1;
+        if (!reviewed) {
+            id = socialSvc.publish(reviewJSON);
+        }
         result.id = id;
-        result.success = (id > -1) ? true : false;
+        result.success = (id > -1);
+        var myReview = JSON.parse(String(socialSvc.getUserComment(username, target)));
+        if (!isEmpty(myReview)) {
+            myReview.actor.id = cleanUsername(myReview.actor.id);
+        } else {
+            myReview = false;
+        }
+        result.myReview = myReview;
         return result;
     };
     ReviewUtils.removeUserReview = function(reviewId, username) {
```

Comment:
Thanks @thushara35 , I have fixed it in [this commit](https://github.com/wso2/carbon-store/pull/844/commits/9ecf5b5091edf9da96e65ce1ed5e68edd3a5704e)